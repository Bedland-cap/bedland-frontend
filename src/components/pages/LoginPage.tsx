import { useNavigate } from 'react-router-dom';
import ChildBox from 'components/pages/Page.styled';
import { PageI } from 'utils/interfaces/PageI';

const LoginPage = (props: PageI) => {
  const nav = useNavigate();

  return (
    <ChildBox theme={{ color: '#131f3e' }}>
      <button
        type='button'
        style={{ width: '50px', height: '50px' }}
        onClick={() => nav('/dashboard')}
      >
        Click me
      </button>
    </ChildBox>
  );
};

export default LoginPage;
