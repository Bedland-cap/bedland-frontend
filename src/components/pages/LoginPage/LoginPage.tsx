import Image from 'components/atoms/Image/Image';
import DefaultImage from 'assets/Loginlogo.svg';
import LoginForm from 'components/organisms/LoginForm/LoginForm';
import SectionRow from 'components/templates/SectionRow/SectionRow';

const LoginPage = () => (
  <SectionRow>
    <Image src={DefaultImage} alt='' />
    <LoginForm />
  </SectionRow>
);
export default LoginPage;
