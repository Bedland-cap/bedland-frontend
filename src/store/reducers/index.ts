import { userReducer } from './user/user_slice';
import { residentReducer } from './resident/resident_slice';
import { buildingReducer } from './building/building_slice';
import { flatReducer } from './flat/flat_slice';

export { buildingReducer, residentReducer, userReducer, flatReducer };
