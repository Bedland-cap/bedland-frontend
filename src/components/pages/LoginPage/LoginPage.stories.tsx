import type { Meta, StoryObj } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import ThemeProvider from 'theme/ThemeContext';
import GlobalStyle from 'theme/globalStyles';

import LoginPage from './LoginPage';

const meta = {
  title: 'pages/LoginPage',
  component: LoginPage,
  tags: ['autodocs'],
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        <BrowserRouter>{Story()}</BrowserRouter>
      </ThemeProvider>
    ),
  ],
} satisfies Meta<typeof LoginPage>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};
