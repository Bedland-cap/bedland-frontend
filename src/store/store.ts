import { combineReducers, configureStore, type PreloadedState } from '@reduxjs/toolkit';
import { CurriedGetDefaultMiddleware } from '@reduxjs/toolkit/dist/getDefaultMiddleware';
import loggerMiddleware from 'redux-logger';
import { userApi } from 'services/user/userApi';
import { flatApi } from 'services/flat/flatApi';
import { managerApi } from 'services/manager/managerApi';
import { residentReducer, buildingReducer, userReducer, flatReducer } from './reducers/index';
import { managerReducer } from './reducers/manager/manager_slice';

export const rootReducer = combineReducers({
  user: userReducer,
  resident: residentReducer,
  manager: managerReducer,
  building: buildingReducer,
  flat: flatReducer,
  [userApi.reducerPath]: userApi.reducer,
  [flatApi.reducerPath]: flatApi.reducer,
  [managerApi.reducerPath]: managerApi.reducer,
});

// Adding the api middleware enables caching, invalidation, polling,
// and other useful features of `rtk-query`.
export const rootMiddleware = (getDefaultMiddleware: CurriedGetDefaultMiddleware) =>
  getDefaultMiddleware().concat(loggerMiddleware, userApi.middleware, flatApi.middleware, managerApi.middleware);

export type RootState = ReturnType<typeof rootReducer>;

const storeInit = (preloadedState?: PreloadedState<RootState>) =>
  configureStore({
    reducer: rootReducer,
    middleware: rootMiddleware,
    preloadedState,
  });

export const reduxStore = storeInit();

export type AppStore = ReturnType<typeof storeInit>;
export type AppDispatch = AppStore['dispatch'];
