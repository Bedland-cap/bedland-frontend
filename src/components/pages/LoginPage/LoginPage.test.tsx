import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import ThemeProvider from 'theme/ThemeContext';

import { Provider } from 'react-redux';
import { reduxStore } from 'store/store';
import LoginPage from './LoginPage';

describe('Login Page', () => {
  it('renders correctly', () => {
    const snapshot = renderer
      .create(
        <ThemeProvider>
          <BrowserRouter>
            <Provider store={reduxStore}>
              <LoginPage />
            </Provider>
          </BrowserRouter>
        </ThemeProvider>,
      )
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
