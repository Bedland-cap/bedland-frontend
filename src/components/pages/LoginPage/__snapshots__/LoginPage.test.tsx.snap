// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page renders correctly 1`] = `
<div
  className="sc-pyfCe hLiJWN"
>
  <img
    alt=""
    className="sc-bcXHqe bSBJnN"
  />
  <form
    className="sc-iBYQkv qrkfQ"
    onSubmit={[Function]}
  >
    <h2
      style={
        {
          "color": "#FCFEFF",
          "fontFamily": "Poppins, sans-serif",
          "fontSize": "2rem",
          "fontWeight": 600,
          "letterSpacing": "2px",
          "lineHeight": 1.2,
          "textAlign": "center",
        }
      }
    >
      Login
    </h2>
    <div
      className="sc-jSUZER igHxvv"
    >
      <label
        className="sc-eDvSVe kvGEhv"
        htmlFor="login"
        size="medium"
      >
        Login
      </label>
      <input
        className="sc-hLBbgP kgjgay"
        id="login"
        name="login"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter your login"
        type="text"
      />
    </div>
    <div
      className="sc-jSUZER igHxvv"
    >
      <label
        className="sc-eDvSVe kvGEhv"
        htmlFor="password"
        size="medium"
      >
        Password
      </label>
      <input
        className="sc-hLBbgP kgjgay"
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter your password"
        type="password"
      />
    </div>
    <span
      className="sc-gswNZR XsqEV"
    />
    <button
      className="sc-gKPRtg liKPGY"
      type="submit"
    >
      Login
    </button>
    <a
      className="sc-ftTHYK caMiFM"
      href="/resetPassword"
      onClick={[Function]}
      palette={
        {
          "accent": "#0E172D",
          "background": "#0A1122",
          "backgroundGrey": "#F5F5F5",
          "badge": "#CF0000",
          "danger": "#FF4438",
          "default": "#000",
          "footer": "#3F4476",
          "greyDark": "#767676",
          "greyLight": "#F5F6F8",
          "inputGrey": "#4B4B4B",
          "light": "#FCFEFF",
          "primaryDark": "#4553DD",
          "primaryLight": "#909AFF",
          "secondaryDark": "#EF6517",
          "secondaryLight": "#FF7C33",
          "shadow": "#4B4D62",
          "sidebarTabs": "#CACACA",
          "success": "#83E57A",
          "successDeep": "#3A9832",
          "widgetBackground": "#fff",
        }
      }
    >
      Forgot your password?
    </a>
  </form>
</div>
`;
